### PR TITLE
authz for search should support GET/POST

### PR DESCRIPTION
**Pull request by jamesc**
- Search should all GET/POST to all.
- Cleaned up some funheads for handle_auth_info/3

### CODE REVIEW

File: src/chef_wm_base.erl [hunk 1]
```diff
@@ -772,21 +773,13 @@ handle_auth_info(Module, Req, _State)
         _Else ->
             forbidden
     end;
-handle_auth_info(Module, Req, _State)
-        when Module =:= chef_wm_depsolver ->
+handle_auth_info(chef_wm_depsolver, Req, _State) ->
```
Comment (seth) [inline src/chef_wm_base.erl:776]: search needs to allow POST for partial search as well.

Comment (jamesc) [inline src/chef_wm_base.erl:776]: It does, at L758-L759 aove

Comment (seth) [inline src/chef_wm_base.erl:776]: Yep. Sorry, missed it in the diff. So if you've verified this with knife or similar, I'd say let's merge it. We'll circle back to refactor it later.

Comment (jamesc) [inline src/chef_wm_base.erl:776]: Yeah, I now see the same ej bad_arith error inside the search endpoint so at least authz is working now

**Pull request merged**